src/lib/authMiddleware.ts

```diff
@@ -39,6 +40,39 @@ const refreshToken = async (
     }
 };
 
+const resetPassword = async (
+    options: AuthistOptions,
+    email?: string
+): Promise<[BearerError | undefined, { token: any } | undefined]> => {
+    if (!email) {
+        return [new NotAuthenticated(ERROR_CODE.EmailRequired), undefined];
+    }
+    try {
+        const token = await resetUserPassword(options)(email);
+        return [undefined, token];
+    } catch (error) {
+        return [error, undefined];
+    }
+};
+
+const changePassword = async (
+    options: AuthistOptions,
+    data?: { token?: any; password?: string }
+): Promise<[BearerError | undefined, undefined]> => {
+    if (!data?.token) {
+        return [new NotAuthenticated(ERROR_CODE.ResetPasswordTokenRequired), undefined];
+    }
+    if (!data?.password) {
```

Comment:
This way we disallow an empty string. I know empty string string does not meet general criteria for safe passwords, but it shouldn't be our call to handle it. We should accept any string that is sent imo.

Reply:
ok like this: `!data?.password && typeof data.password !== 'string'`